man/coap_block.txt.in

```diff
@@ -131,15 +172,27 @@ This function must only be called once per _pdu_.  When the final block is
 transmitted, the callback function _release_func_ (if not NULL) with the user
 defined parameter of _app_ptr_ is called so that the data can be released.
 
+*NOTE:* _data_ must not be a local variable on the stack if the function
+calling *coap_add_data_large_request*() returns, otherwise _data_
+will potentially get corrupted when the stack space for _data_ is re-used by
+another function. Subsequent packets sent for remainder of _data_ could then
+end up containing unexpected data.
+
 The *coap_add_data_large_response*() function is responsible for handling
 the server's large responses to requests. *coap_add_data_large_response*()
 should be used as a direct replacement for *coap_add_data*() if it is possible
 that the _length_ of _data_ will not fit into a single server's response pdu.
 This function adds in the initial part of the payload _data_ of length
 _length_ to the PDU _pdu_. _release_func_ (if not NULL) and _app_ptr_ are
-used for releasing the data when the body transfer is complete.  It also adds
-in the appropriate CoAP options such as BLOCK2, SIZE2 and ETAG to handle
-Block-Wise transfer if the data does not fit into a single PDU.
+used for releasing the data when the body transfer is complete.
+
+*NOTE:* _data_ must not be a local variable on the stack, otherwise _data_
+will potentially get corrupted when the request handler returns and other
+functions then re-use the same stack space. Subsequent packets sent for
+remainder of _data_ could then end up containing unexpected data.
+
+It also adds in the appropriate CoAP options such as BLOCK2, SIZE2 and ETAG to
+handle Block-Wise transfer if the data does not fit into a single PDU.
```

Comment:
"Block-Wise" → "block-wise"

"fit into" → "fit in"

Reply:
Updated.  Left it as Block-Wise where the RFC7959 title is referred to.

Other "fit into" corrected.  Code changes pushed.